src/FilterChain.php

```diff
@@ -178,7 +178,7 @@ public function attach($callback, $priority = self::DEFAULT_PRIORITY)
                     get_debug_type($callback)
                 ));
             }
-            $callback = [$callback, 'filter'];
+            $callback = $callback->filter(...);
```

Comment:
This causes test failures in `laminas/input-filter` - However, the test case over there is wrong to assert an array when the return type is clearly documented as `callable`

Reply:
Ref: https://github.com/laminas/laminas-inputfilter/pull/118